src/CodeAnalyzer.py

```diff
@@ -0,0 +1,88 @@
+stack_protectors = ["__stack_chk_fail", "___stack_chk_guard", "in_FS_OFFSET"]
+
+class CodeAnalyzer:
+    def __init__(self, signature, func_code):
+        self.func_code = func_code
+        self.signature = signature
+        self.types_of_variables = {}
+        
+    def get_variable_name(self, variable):
+        if "*" in variable:
+                variable = variable[variable.rfind("*")+1:]
+        if "[" in variable:
+            variable = variable[:variable.find("[")]
+        if ";" in variable:
+            variable = variable[:variable.find(";")]
+        return variable
+    
+    def get_variables_types_from_signature(self):
+        for argument in self.signature[self.signature.find("(")+1:self.signature.find(")")].split(","):
+            if argument == "void":
+                break
+            type, variable = argument.split()
+            variable = self.get_variable_name(variable)
+            self.types_of_variables[variable] = type
+        return self.types_of_variables
+
+    def remove_warnings(self, lines, id, warning_in_load, warning_in_store):
+        lvalue, rvalue = lines[id].split(" = ")
+        variable = self.get_variable_name(lvalue.strip())
+        if warning_in_load:
+            if rvalue[1] == "(":
+                lines[id] = lvalue + " = " + rvalue[0] + f"({self.types_of_variables[variable]} *)" + rvalue[rvalue.index(")")+1:]
+            else:
+                lines[id] = lvalue + " = " + rvalue[0] + f"({self.types_of_variables[variable]} *)" + rvalue[1:]
+        
+        if warning_in_store:
+            if lvalue[1] == "(":
+                lines[id] = lvalue[0] + f"({self.types_of_variables[variable]} *)" + lvalue[lvalue.index(")")+1:] + " = " + rvalue
+            else:
+                lines[id] = lvalue[0] + f"({self.types_of_variables[variable]} *)" + lvalue[1:] + " = " + rvalue
+
+    def get_code_without_warnings(self):
+        warning_in_load = False
+        warning_in_store = False
+        need_to_delete_brace = False
+
+        self.types_of_variables = self.get_variables_types_from_signature()
+
+        lines = self.func_code.split("\n")
+        declarationID = 0
+        while "{" not in lines[declarationID]:
+            declarationID += 1
+        declarationID += 1
+
+        declaration = lines[declarationID][:lines[declarationID].find(" [")].split()
+        while len(declaration) == 2 and declaration[0] not in ["return", "do"]:
+            type, variable = declaration[0], self.get_variable_name(declaration[1])
+            self.types_of_variables[variable] = type
+            declarationID += 1
+            declaration = lines[declarationID][:lines[declarationID].find(" [")].split()
```

Comment:
```suggestion
        line_ID = 0
        while "{" not in lines[line_ID]:
            line_ID += 1
        line_ID += 1

        line = lines[line_ID][:lines[line_ID].find(" [")].split()
        while len(line) == 2 and line[0] not in ["return", "do"]:
            type, variable = line[0], self.get_variable_name(line[1])
            self.types_of_variables[variable] = type
            line_ID += 1
            line = lines[line_ID][:lines[line_ID].find(" [")].split()
```
Мне кажется лучше назвать line, line_ID, или codeline и codeline_ID, т.к. в строке может быть что угодно (не только return и do, но и if, LABEL, например), и название declaration сбивает с толку